src/cli/rustup_mode.rs

```diff
@@ -1340,14 +1340,17 @@ fn target_remove(cfg: &Cfg, m: &ArgMatches) -> Result<utils::ExitCode> {
         if target == default_target {
             warn!("after removing the default host target, proc-macros and build scripts might no longer build");
         }
-        let has_at_most_one_target = {
-            let components = distributable.components()?;
-            // Every component target that is not `None` (wildcard).
-            let targets = components
-                .iter()
-                .filter_map(|c| c.installed.then(|| c.component.target.clone()).flatten());
-            targets.unique().at_most_one().is_ok()
-        };
+        // Whether we have at most 1 component target that is not `None` (wildcard).
+        let has_at_most_one_target = distributable
+            .components()?
+            .into_iter()
+            .filter_map(|c| match (c.installed, c.component.target) {
+                (true, Some(t)) => Some(t),
+                _ => None,
+            })
```

Comment:
Maybe this would be even better?

```suggestion
            .filter(|c| c.installed)
            .filter_map(|c| c.component.target)
```

Reply:
Personally I prefer my variant: it seems confusing to me (and potentially less efficient) to run two closures in subsequent combinators, when these could just be combined into a single function.

Reply:
@djc Thanks for your opinion! I'll stick with your version then.